docs/src/darray.md

```diff
@@ -0,0 +1,135 @@
+ 0.805897   0.0275489  0.175715   0.135956   0.389958    0.856349      0.974141   0.586308   0.59695     0.906727
+ 0.212875   0.509612   0.85531    0.266659   0.0695836   0.0551129     0.788085   0.401581   0.948216    0.00242077
+ 0.512997   0.134833   0.895968   0.996953   0.422192    0.991526   …  0.838781   0.141053   0.747722    0.84489
+ 0.283221   0.995152   0.61636    0.75955    0.072718    0.691665      0.151339   0.295759   0.795476    0.203072
+ 0.0946639  0.496832   0.551496   0.848571   0.151074    0.625696      0.673817   0.273958   0.177998    0.563221
+ 0.0900806  0.127274   0.394169   0.140403   0.232985    0.460306      0.536441   0.200297   0.970311    0.0292218
+ 0.0698985  0.463532   0.934776   0.448393   0.606287    0.552196      0.883694   0.212222   0.888415    0.941097
+```
+
+An existing array can be distributed by using `Distribute`.
+```julia
+julia> Z = zeros(100, 500);
+
+julia> Dzeros = Distribute(Blocks(10, 50), Z)
+Distribute{Float64, 2}(100, 500)
+
+julia> compute(Dzeros)
+Dagger.DArray{Any, 2, typeof(cat)}(100, 500)
+```
+
+## Using Distributed Arrays
+
+Arithmetic using `DArray` and related types are done lazily. See the following example.
+
+```julia
+julia> using Distributed; addprocs(6)
+6-element Vector{Int64}:
+ 2
+ 3
+ 4
+ 5
+ 6
+ 7
+
+julia> using Dagger
+
+julia> DX = rand(Blocks(50,50), 100, 100)
+Dagger.AllocateArray{Float64, 2}(100, 100)
+
+julia> DY = DX + DX
+Dagger.BCast{Base.Broadcast.Broadcasted{Dagger.DaggerBroadcastStyle, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}, typeof(+), Tuple{Dagger.AllocateArray{Float64, 2}, Dagger.AllocateArray{Float64, 2}}}, Float64, 2}(100, 100)
+
+julia> DZ = DY .* 3
+Dagger.BCast{Base.Broadcast.Broadcasted{Dagger.DaggerBroadcastStyle, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}, typeof(*), Tuple{Dagger.BCast{Base.Broadcast.Broadcasted{Dagger.DaggerBroadcastStyle, Tuple{Base.OneTo{Int64}, Base.OneTo{Int64}}, typeof(+), Tuple{Dagger.AllocateArray{Float64, 2}, Dagger.AllocateArray{Float64, 2}}}, Float64, 2}, Int64}}, Float64, 2}(100, 100)
+
+julia> size(DZ)
+(100, 100)
+
+julia> DA = compute(DZ)
+Dagger.DArray{Any, 2, typeof(cat)}(100, 100)
+
+julia> Dagger.chunks(DA)
+2×2 Matrix{Union{Thunk, Dagger.Chunk}}:
+ Chunk{Matrix{Float64}, DRef, ThreadProc, AnyScope}(Matrix{Float64}, ArrayDomain{2}((1:50, 1:50)), DRef(4, 8, 0x0000000000004e20), ThreadProc(4, 1), AnyScope(), true)  …  Chunk{Matrix{Float64}, DRef, ThreadProc, AnyScope}(Matrix{Float64}, ArrayDomain{2}((1:50, 1:50)), DRef(2, 5, 0x0000000000004e20), ThreadProc(2, 1), AnyScope(), true)
+ Chunk{Matrix{Float64}, DRef, ThreadProc, AnyScope}(Matrix{Float64}, ArrayDomain{2}((1:50, 1:50)), DRef(5, 5, 0x0000000000004e20), ThreadProc(5, 1), AnyScope(), true)     Chunk{Matrix{Float64}, DRef, ThreadProc, AnyScope}(Matrix{Float64}, ArrayDomain{2}((1:50, 1:50)), DRef(3, 3, 0x0000000000004e20), ThreadProc(3, 1), AnyScope(), true)
+
```

Comment:
Maybe a brief mention that these "chunks" now point to the actual result of the above calculations, split into blocks, but that users don't typically need to worry about these details?